src/hooks/useNotificationSetup.ts

```diff
@@ -1,11 +1,26 @@
-import { useEffect } from 'react'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import { useLinkTo } from '@react-navigation/native'
+import { useCallback, useEffect, useState } from 'react'
 import { Alert } from 'react-native'
 
+import { useAuth } from './useAuth'
+
+import { ASYNC_STORAGE_KEYS } from '~constants'
 import { useNotificationContext } from '~contexts'
+import { isAuthorizedLink } from '~navigation/linking'
 import { registerForPushNotificationsAsync } from '~services'
 
-export const useNotificationSetup = () => {
+type Options = {
+  /**
+   * When *true* automatically handles deeplink provided in notification.
+   */
+  enableDeeplink?: boolean
+}
+export const useNotificationSetup = (opts?: Options) => {
+  const [deeplink, setDeeplink] = useState<string>()
```

Comment:
Why do we need this, if you don't use it later any where?

Reply:
Just to extend hook API - I had in mind that for some users it might be useful to handle deeplinks manually so I let them use it as deeplink was returned from that hook.

Reply:
But in the end, I removed that 😅